### PR TITLE
Console upgrade

### PR DESCRIPTION
**Pull request by campagnola**
An overhaul of ConsoleWidget:
- Adds proper handling / display of chained exceptions
- fixes some issues preventing correct display of exception tracebacks
- refactors ConsoleWidget into several independent components (I was hoping to be able to swap in a readline-like console experience, but no luck so far)
- minor GUI tweaks
- added an example showing exception handling features
- updated main example list

### CODE REVIEW

File: pyqtgraph/examples/console_exception_inspection.py [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+    x = "inside runInStack4(func)"
+    return func()
+
+
+class SignalEmitter(pg.QtCore.QObject):
+    signal = pg.QtCore.Signal(object, object)
+    def __init__(self, queued):
+        pg.QtCore.QObject.__init__(self)
+        if queued:
+            self.signal.connect(self.run, pg.QtCore.Qt.QueuedConnection)
```
Comment (pijyoi) [inline pyqtgraph/examples/console_exception_inspection.py:111]: `QtCore.Qt.ConnectionType.QueuedConnection`

File: pyqtgraph/console/Console.py [hunk 2]
```diff
@@ -45,450 +41,138 @@ def __init__(self, parent=None, namespace=None, historyFile=None, text=None, edi
...
-        # send exceptions raised in non-gui threads back to the main thread by signal.
-        self._threadException.connect(self._threadExceptionHandler)
-        
+    def _setupUi(self):
+        self.layout = QtWidgets.QGridLayout(self)
+        self.setLayout(self.layout)
+        self.layout.setContentsMargins(0, 0, 0, 0)
+        self.layout.setSpacing(0)
+
+        self.splitter = QtWidgets.QSplitter(QtCore.Qt.Vertical, self)
```
Comment (pijyoi) [inline pyqtgraph/console/Console.py:80]: `QtCore.Qt.Orientation.Vertical`

**Comment by j9ac9k** [comment]
CI says "yes" now; anything else you want to add or is this PR reflect the state you want to merge?

**Comment by campagnola** [comment]
Looks done to me!

**Comment by j9ac9k** [comment]
Would you mind addressing the unused variables and import in `pyqtgraph/examples/console_exception_inspection.py` so codeql doesn't have a cow?

**Comment by campagnola** [comment]
> Would you mind addressing the unused variables and import in `pyqtgraph/examples/console_exception_inspection.py` so codeql doesn't have a cow?

Will CodeQL continue having cows even after the code is merged? The variables it's complaining about actually _are_ used; CodeQL just can't tell.

**Comment by campagnola** [comment]
The `x` variables are just there to give the user something to inspect when an exception is caught. I'll add returns to make CodeQL happy..

**Comment by j9ac9k** [comment]
Merging, thanks @campagnola

**Pull request merged**